### PR TITLE
Puzzles don't show bubbles if already solved

### LINKED ISSUE #199: Don't need bubbles when reloading a puzzle that has already been solved.
* If loading a puzzle that has already been solved, no need to show bubbles. Just highlight constraints as green!
* Requested by several players (e.g., malcolm at EternaCon 2018)
* Looks like behavior got worse in transition from FLASH to Typescript -- previously 'space bar puzzles' at least required space bar before bubbles appeared.

### PR DESCRIPTION
**Pull request by AlexHalb**
Resolves #199

### CODE REVIEW

File: src/eterna/puzzle/PuzzleManager.ts [hunk 1]
```diff
@@ -412,6 +412,12 @@ export default class PuzzleManager {
 
         let puzzle = await this.parsePuzzle(data['puzzle']);
 
+        let cleared = data.cleared as { nid: string }[];
+        let clearedNIDs = cleared.map((e) => e.nid);
+        if (clearedNIDs.some((e) => +e === puzzle.nodeID)) {
```
Comment (luxaritas) [inline src/eterna/puzzle/PuzzleManager.ts:417]: Maybe use parseInt instead?

**Comment by luxaritas** [comment]
Sorry, one more tweak now that I'm seeing how this behaves. Instead of showing the end screen, just show the arrows at the bottom. The best way to approach this may be to just automatically hide the mission complete screen as soon as you go to show it. Basically the issue here is that it's still being disruptive

**Comment by luxaritas** [comment]
It looks like it still submits the solution - can we bypass that? I imagine that the next puzzle button will need to be omitted, but that's fine.

Also getting an error in the console at that point:
```
Uncaught (in promise) TypeError: missionClearedPanel is null
    showMissionClearedPanel PoseEditMode.ts:2170
    _callee2$ PoseEditMode.ts:2034
    Babel 12
    submitCurrentPose PoseEditMode.ts:1834
    updateScore PoseEditMode.ts:2627
    setToTargetMode PoseEditMode.ts:1741
    changeTarget PoseEditMode.ts:1567
    onSetPip PoseEditMode.ts:1447
    setPip GameMode.ts:224
    setPuzzleEpilog PoseEditMode.ts:2464
    _poseEditByTargetCb PoseEditMode.ts:988
    update PoseEditMode.ts:1356
```

**Comment by AlexHalb** [comment]
The solution submission can be bypassed, but at the cost of almost everything in the mission cleared panel (rank scroll, next puzzle button, description, etc). If we're going to leave that out, does it make more sense just to leave out the arrow entirely?

**Comment by AlexHalb** [comment]
I suppose a potential solution could be to send the next puzzle with the puzzle (we already do this, but it's null for all cases I've seen) as well as the current user's rank (so a stationary rank scroll could be created).

**Comment by luxaritas** [comment]
To be honest, the rank scroll and next button are largely irrelevant in this use case. What I care about is the mission complete text (the message + more info, or whatever they're called), which are provided at puzzle init anyways

**Pull request merged**